frontend/src/app/containers/App.js

```diff
@@ -100,14 +104,51 @@ class App extends Component {
       }
     });
     this.measurePageview();
+
+    const langs = {};
+
+    function addLang(lang, response) {
+      if (response.ok) {
+        response.text().then(data => {
+          langs[lang] = `${langs[lang] || ''}${data}
+`;
+        });
+      }
+    }
+
+    Promise.all(navigator.languages.map(language =>
+      Promise.all(
+        [
+          fetch(`/static/locales/${language}/app.ftl`).then(response => addLang(language, response)),
+          fetch(`/static/locales/${language}/experiments.ftl`).then(response => addLang(language, response))
+        ]
+      )
+    )).then(() => this.props.setLocalizations(langs));
```

Comment:
You could possibly store the entire `messages` iterator in the state, instead of raw localization files, which would make the code in `render` a bit simpler.  You also wouldn't need the `addLang` function; you can use `cx.addMessages(data);` multiple times on a single `MessageContext` instance.